components/fieldlabel/stories/fieldlabel.stories.js

```diff
@@ -39,15 +41,91 @@ export default {
 		alignment: "left",
 		isDisabled: false,
 		isRequired: false,
+		label: "Label",
 	},
 	parameters: {
 		packageJson: pkgJson,
 	},
 };
 
+/**
+ * By default, a field label has left-aligned text, and is a medium size. For field label text, use a short, catch-all description (1-3 words) of the information that a user needs to provide.
+ * 
+ * The default position of a field label is above an input, but it can alternatively be positioned to the side. Visit the [form component](/docs/components-form--docs) to see examples of the field label with an input.
+ 
+ */
 export const Default = FieldLabelGroup.bind({});
-Default.args = {
+Default.args = {};
+
+// ********* DOCS ONLY ********* //
+/**
+ * Field labels come in four different sizes: small, medium, large, and extra-large. The medium size is the default and most frequently used option with medium-sized inputs. Use the other sizes sparingly; they should be used to create a hierarchy of importance within the page. 
+ * 
+ * Both small and medium field labels have the same font size, but different paddings when used as side labels.
+ */
+export const Sizing = (args, context) => Sizes({
+	Template,
+	withBorder: false,
+	withHeading: false,
+	...args,
+}, context);
+Sizing.tags = ["!dev"];
+Sizing.parameters = {
+	chromatic: { disableSnapshot: true },
+};
+
+/**
+ * To render right-aligned field label text, apply the `.spectrum-FieldLabel--right` class to the field label.
+ * 
+ */
+export const RightAligned = Template.bind({});
+RightAligned.args = {
 	label: "Label",
+	alignment: "right",
+	customStyles: {
+		width: "72px",
+	},
+};
+RightAligned.tags = ["!dev"];
+RightAligned.parameters = {
+	chromatic: { disableSnapshot: true },
+};
+RightAligned.storyName = "Right-aligned";
+
+/**
+ * Field labels for required inputs can be marked with an asterisk at the end of the label. Optional inputs would then be understood to not have the asterisk. If using the asterisk icon, do not leave any space between the label text and the start of the `<svg>` element in the markup. Extra space should not be added in addition to the inline margin.
+ */
+export const Required = Template.bind({});
+Required.args = {
+	isRequired: true,
+};
+Required.tags = ["!dev"];
+Required.parameters = {
+	chromatic: { disableSnapshot: true },
+};
+
+/**
+ * When the field label is too long for the available horizontal space, it wraps to form another line.
+ */
+export const WrappingAndRequired = Template.bind({});
```

Comment:
I don't mind it on the docs page, it certainly doesn't hurt to over-communicate in this case!

Reply:
This is an opinion but I wouldn't mind a control or something (like for instance restricting the width) in Storybook that would allow me to see how Field label wraps, not strictly necessary but maybe nice to have because otherwise the label can hold a lot of text before it'll start to wrap:
![image](https://github.com/user-attachments/assets/9897aef6-ae1a-4953-9c61-e583d85ee4d5)

Reply:
Hmm...this is a difficult one for me. It's definitely a little odd that this poor field label is just hanging out by itself, out of context. And I suppose that's my only hesitancy with adding a control, and we can use `customStyles` if needed (not that `customStyles` is shown on the docs page or in the controls table). Usually, a field label will be inside of some sort of container or form or something, right? If I go to textfield, the field label wraps at the width of the text field container. The wrapping doesn't seem to be an issue.

<img width="267" alt="Screenshot 2024-10-01 at 5 23 15 PM" src="https://github.com/user-attachments/assets/e78ef7a6-68ce-48cb-8ed9-f48a35d76fa4">

I don't have label controls set up for form right now, but if I go into the form.stories.js file and make up a long label, I don't actually see the text wrapping unless I use a ton of text like you did, I change the screen size, or again, pass some `customStyles` to form.

<img width="904" alt="Screenshot 2024-10-01 at 5 29 29 PM" src="https://github.com/user-attachments/assets/2d035bb2-b704-4950-ac04-522179b50003">

My gut says we don't need a control for this (namely because this component is just out of context and probably shouldn't be used on its own), but I'm open and can totally add it in if you think it's valuable.

Reply:
I could also add something like this to the `WrappingAndRequired` story docs: 

"Implementations can pass a maximum width to the field label component using `customStyles`."